src/dict.c

```diff
@@ -1173,7 +1173,8 @@ void dictGetStats(char *buf, size_t bufsize, dict *d) {
 
 #ifdef REDIS_TEST
 
-#include "sds.h"
+#include <sdscompat.h>
+#include <sds.h>
```

Comment:
this can cause some issues, bringing hiredis's sds into redis-server.
IIUC, the problem is that when building redis-cli, we take the dict.o file that was compiled with `-DREDIS_TEST`.
i think this is an indication that we need to do one of the following:
1. avoid using dict.c in redis-cli (obviously not an option).
2. compile 2 different copies of dict.o, the one used by redis-cli should be one that doesn't have `-DREDIS_TEST` (a bit complicated)
3. avoid using sds in the dict.c test. (this is probably not too hard)
4. decide that the dict-benchmark is not actually a test to be part of the REDIS_TEST campaign (sounds logical, this is actually a benchmark, not a test)

i think the easiest thing is to go with 4, and just revert the changes to the Makefile and dict.c

Reply:
@oranagra I try to go with 3.